### PR TITLE
Edit fixes

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers 

This fixes the following on editing:
1. Escape key works
2. If you edit you get a little area on the left that tells you you're editing
3. Edited message and input box now have a yellow background 
4. Scrolls to the editing message (kinda janky sometimes)
5. @mentions work on edited message
6. Editing text isn't initially all selected
7. Editing message is slightly transparent w/ yellow

cc: @patrickxb @cecileboucheron @mmaxim @malgorithms

### CODE REVIEW

**Comment by chrisnojima** [comment]
@cecileboucheron the editing 'tab' thing can obscure the message if you're editing the very last message so we obvs need to move it / something

**Comment by chrisnojima** [comment]
i'm removing the changes to editing map

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready again

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/chat/conversation/input-area/normal/container.js [hunk 1]
```diff
@@ -163,11 +167,15 @@ export default compose(
...
-        const i = this.props.inputGetRef()
-        // Might be a better way to do this but this is simple for now
-        setImmediate(() => {
-          i && i.select()
-        })
+
+        if (!isMobile) {
+          const i = this.props.inputGetRef()
+          setImmediate(() => {
+            if (i) {
```
Comment (buoyad) [inline shared/chat/conversation/input-area/normal/container.js]: can you move `if (i)` to before the `setImmediate`?

File: shared/chat/conversation/input-area/normal/index.desktop.js [hunk 2]
```diff
@@ -255,6 +272,14 @@ const EmojiPicker = ({emojiPickerToggle, onClick}) => (
   </Box>
 )
 
+const editingTabStyle = {
+  ...globalStyles.flexBoxColumn,
+  alignItems: 'flex-start',
+  backgroundColor: globalColors.yellow_60,
+  height: 48,
```
Comment (buoyad) [inline shared/chat/conversation/input-area/normal/index.desktop.js]: Looks a little weird when editing a long message
image

File: shared/chat/conversation/list-area/normal/container.js [hunk 3]
```diff
@@ -11,24 +12,32 @@ import {
...
   return {
     conversationIDKey,
+    editingOrdinal: state.chat2.editingMap.get(conversationIDKey),
     hasExtraRow,
     messageOrdinals: Constants.getMessageOrdinals(state, conversationIDKey),
   }
 }
 
-const mapDispatchToProps = (dispatch: Dispatch, {conversationIDKey}) => ({
+type DispatchProps = {
```
Comment (buoyad) [inline shared/chat/conversation/list-area/normal/container.js]: Why type this?

Comment (chrisnojima) [reply shared/chat/conversation/list-area/normal/container.js]: flow wasn't typing it

**Pull request merged**